GWAS/LMM.ipynb

```diff
@@ -1208,6 +1209,156 @@
+    "    -bfile ${_input:n} \\\n",
+    "    -gk 2 \\\n",
+    "    -o ${_output:bnn}\n",
+    "bash: container=container_lmm,expand='${ }', stderr = f'{_output[0]:n}.stderr', stdout = f'{_output[0]:n}.stdout'\n",
+    "    cp ./output/${bfile:bn}.sXX.txt ${cwd}\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "source": [
+    "### Step 2 :Fitting the null"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "outputs": [],
+   "source": [
+    "# Run GMMAT step1: fit a GLMM with covariate adjustment and random effects to account for population structure and family or cryptic relatedness\n",
+    "[GMMAT_1]\n",
+    "#use the standardized GRM file generated in gemma steps\n",
+    "parameter: grmFile = path(f'{cwd}/{bfile:bn}.sXX.txt')\n",
+    "#a colum in the  data frame data, indicaing e id of samples\n",
+    "parameter: phenoCol = 'AD'\n",
+    "parameter: idCol = 'IID'\n",
+    "input: phenoFile, f'{bfile:n}.fam', grmFile\n",
+    "output: f'{cwd}/{bfile:bn}.{phenoFile:bn}.GMMAT.rds'\n",
+    "R: container=container_lmm,expand='${ }', stderr = f'{_output[0]:n}.stderr', stdout = f'{_output[0]:n}.stdout'\n",
+    "    library('GMMAT')\n",
+    "    library('data.table')\n",
+    "    library('dplyr')\n",
+    "   \n",
+    "    #Prepare phenotype and covariates in an R data frame\n",
+    "    pheno = data.table::fread(${_input[0]:r}, header = TRUE)\n",
+    "    #Pheno are currently coded as 1, 2, and -9, need to recoded as 0,1,and NA\n",
+    "    pheno$${phenoCol}=  recode(pheno$${phenoCol}, `2` = 1, `1` = 0, `-9` = NULL)\n",
+    "    #Prepare GRM file in a R data frame\n",
+    "    GRM = as.matrix(data.table::fread(${_input[2]:r}, header = FALSE))\n",
+    "    #Extract IIDs from .fam file\n",
+    "    id_vector = read.table(${_input[1]:r}, header = FALSE)[,2]\n",
+    "    #make the GRM colnames and rownames using the actual IID\n",
+    "    colnames(GRM) = id_vector\n",
+    "    rownames(GRM) = id_vector\n",
+    "\n",
+    "    fit_null = glmmkin(${phenoCol} ~ ${\"+\".join(covarCol + qCovarCol)} , \n",
+    "                           data = pheno, \n",
+    "                           kins = GRM, \n",
+    "                           id = '${idCol}',\n",
+    "                           family = binomial(link = \"logit\"))\n",
+    "    saveRDS(fit_null, ${_output:r})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "source": [
+    "### Step 3: Perform single variant score test"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "outputs": [],
+   "source": [
+    "#Run GMMAT step2: single variant score test(based on the null model built above)\n",
+    "[GMMAT_2]\n",
+    "#the maximum rate alllowed for a variant to be included\n",
+    "parameter: miss_cutoff = float\n",
```

Comment:
How about we set a default to it? Also can we name the parameter to be consistent with this workflow notebook:

https://github.com/cumc/bioworkflows/blob/master/GWAS/GWAS_QC.ipynb

see `qc_2` step.